client/state/themes/actions/activate-theme.js

```diff
@@ -46,7 +51,24 @@ export function activateTheme(
 				...( isEnabled( 'themes/theme-switch-persist-template' ) && {
 					persist_homepage_template: true,
 				} ),
-				...( styleVariationSlug && { style_variation_slug: styleVariationSlug } ),
+			} )
+			.then( async ( theme ) => {
+				if ( styleVariationSlug ) {
+					const themeStylesheet = theme.stylesheet || themeId;
+					const globalStylesId = await dispatch( getGlobalStylesId( siteId, themeStylesheet ) );
```

Comment:
[nit] We can move this line to:
```js
if ( currentVariation ) {
     const globalStylesId = await dispatch( getGlobalStylesId( siteId, themeStylesheet ) );
     await dispatch( updateGlobalStyles( siteId, globalStylesId, currentVariation ) );
}
```

What do you think?

Reply:
Sounds good 👍